### PR TITLE
Limiter to track limits by chainID

### PR DESCRIPTION
**Pull request by Bridgerz**

### CODE REVIEW

File: bridge/evm/contracts/BridgeLimiter.sol [hunk 1]
```diff
@@ -168,9 +192,19 @@ contract BridgeLimiter is IBridgeLimiter, CommitteeUpgradeable, OwnableUpgradeab
...
+
+    /* ========== PRIVATE FUNCTIONS ========== */
+
+    function getChainHourTimestampKey(uint8 chainID, uint32 hourTimestamp)
+        public
+        pure
+        returns (uint256)
+    {
+        return (uint256(chainID) << 32) | uint256(hourTimestamp);
+    }
```
Comment (longbowlu) [inline bridge/evm/contracts/BridgeLimiter.sol:209]: 1. is this a common way in solidity to substitute a tuple key in mapping? and how much more efficient is this compared to nested mapping?
2. this is not `/* ========== PRIVATE FUNCTIONS ========== */`

File: bridge/evm/test/BridgeLimiterTest.t.sol [hunk 2]
```diff
@@ -148,12 +151,75 @@ contract BridgeLimiterTest is BridgeBaseTest {
...
+        assertEq(limiter.oldestChainTimestamp(11), uint32(block.timestamp / 1 hours));
+        assertEq(limiter.oldestChainTimestamp(12), uint32(block.timestamp / 1 hours));
+
+        // check that limits are checked correctly
+        uint8 tokenID = 3;
+        uint256 amount = 999999 * 1000000; // USDC has 6 decimals
+        assertFalse(limiter.willAmountExceedLimit(11, tokenID, amount));
+        limiter.recordBridgeTransfers(11, tokenID, amount);
+        assertTrue(limiter.willAmountExceedLimit(11, tokenID, 2000000));
+        assertFalse(limiter.willAmountExceedLimit(11, tokenID, 1000000));
```
Comment (longbowlu) [inline bridge/evm/test/BridgeLimiterTest.t.sol:212]: check that chain 12 is clean (not impacted by chain 11)

File: bridge/evm/test/BridgeLimiterTest.t.sol [hunk 3]
```diff
@@ -148,12 +151,75 @@ contract BridgeLimiterTest is BridgeBaseTest {
...
+
+        // check that transfers are recorded correctly
+        amount = 1100000 * 1000000; // USDC has 6 decimals
+        limiter.recordBridgeTransfers(12, tokenID, amount);
+        assertEq(
+            limiter.chainHourlyTransferAmount(
+                limiter.getChainHourTimestampKey(12, uint32(block.timestamp / 1 hours))
+            ),
+            11000000000
```
Comment (longbowlu) [inline bridge/evm/test/BridgeLimiterTest.t.sol:221]: check that chain 11 is clean (not impacted by chain 12)

**Review by longbowlu — APPROVED** [review]

File: bridge/evm/contracts/BridgeLimiter.sol (hunk 1 above)
Comment (Bridgerz) [reply bridge/evm/contracts/BridgeLimiter.sol:209]: 1. Nested mappings are not as gas efficient as the composite key approach. Trade off is it's less readable... both approaches are very common. 
2. Good call

File: bridge/evm/test/BridgeLimiterTest.t.sol (hunk 2 above)
Comment (Bridgerz) [reply bridge/evm/test/BridgeLimiterTest.t.sol:212]: Good call. Will do.

File: bridge/evm/test/BridgeLimiterTest.t.sol (hunk 3 above)
Comment (Bridgerz) [reply bridge/evm/test/BridgeLimiterTest.t.sol:221]: 👍

**Pull request merged**